From 6abb68b2f3c0152c2ad2db15231f1740be4676e3 Mon Sep 17 00:00:00 2001
From: Sebastian Schuberth <sschuberth@gmail.com>
Date: Tue, 26 Aug 2014 15:06:52 +0200
Subject: [PATCH 1/3] Add usage help output

Taken from upstream lndir-1.0.1-2-msys-1.0.13-src.tar.lzma's
lndir-1.0.1-2-msys.patch.

Signed-off-by: Sebastian Schuberth <sschuberth@gmail.com>
---
 lndir.c | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/lndir.c b/lndir.c
index 6a8d950..22fedd7 100644
--- a/lndir.c
+++ b/lndir.c
@@ -72,6 +72,27 @@ int with_revinfo = 0;		/* -withrevinfo */
 char *rcurdir;
 char *curdir;
 
+static void _X_NORETURN
+help (FILE* f, char * prog_name, int code)
+{
+    fprintf(f, "lndir (X.org version 6.8.1.0)\n");
+    fprintf(f, "   Creates a shadow directory of symbolic links to another directory tree.\n\n");
+    fprintf(f, "Usage: %s [OPTION...] fromdir [todir]\n\n", prog_name);
+    fprintf(f, "Options:\n");
+    fprintf(f, "  -silent        Do not print name of each subdirectory traversed\n");
+    fprintf(f, "  -ignorelinks   Do not treak symlinks differerntly from regular files\n");
+    fprintf(f, "  -withrevinfo   Symlink contents of source control directories (CVS, .svn, ...)\n");
+    fprintf(f, "  -h, -?, -help  Show this help message.\n\n");
+    fprintf(f, "Other arguments:\n");
+    fprintf(f, "  fromdir        The directory whose contents should be symlinked.\n");
+    fprintf(f, "                 If a relative path, it should be specified from the perspective\n");
+    fprintf(f, "                 of inside the [todir] directory.\n");
+    fprintf(f, "  todir          Optionally, the directory in which the symlinks to the contents\n");
+    fprintf(f, "                 of fromdir will be created.  Defaults to '.'\n");
+
+    exit (code);
+}
+
 static void _X_ATTRIBUTE_PRINTF(2,3) _X_NORETURN
 quit (int code, const char * fmt, ...)
 {
@@ -339,6 +360,11 @@ main (int argc, char *argv[])
 	    ignore_links = 1;
 	else if (strcmp(*argv, "-withrevinfo") == 0)
 	    with_revinfo = 1;
+	else if ((strcmp(*argv, "-h") == 0) ||
+		(strcmp(*argv, "-?") == 0) ||
+		(strcmp(*argv, "-help") == 0) ||
+		(strcmp(*argv, "--help") == 0))
+	    help (stdout, prog_name, 0);
 	else if (strcmp(*argv, "--") == 0) {
 	    ++argv, --argc;
 	    break;
-- 
1.9.4.msysgit.1

